keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1237 @@
+
+
+          ```yaml
+          apiVersion: v1
+          kind: Pod
+          ...
+          spec:
+            resources:
+              requests:
+                memory: 100Mi
+                cpu: 1
+              limits:
+                memory: 400Mi
+                cpu: 2.5
+            containers:
+            - name: container-1
+                resources:
+                  requests:
+                    memory: 60Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 300Mi
+                    cpu: 2  
+            - name: container-2
+                resources:
+                  requests:
+                    memory: 40Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 200Mi
+                    cpu: 1 
+          ```
+
+          The pod-level limit will serve as an aggregate ceiling. This will cover the scenario where the sum of peak usage for individual containers in a pod is high, but these
+          peaks don’t necessarily occur simultaneously.
+
+          If we simply set the pod-level limit as the sum of the individual container limits, we might overallocate resources. The pod might never actually use all those resources
+          at once, leading to wasted capacity on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+  3. Validation logic comparing requests and limits
+
+      1. Pod level requests greater than pod level limits is invalid.
+
+      2. Pod level requests greater than sum of limits of all containers is invalid.
+
+      3. Sum of container level requests greater than Pod level limits is invalid.
+        
+          This is already ensured by other validation rules in 1. and 2. as following:
+        
+          As per 1. Container level Requests <= Pod level requests is valid<br>
+          And as per 3.1 Pod-level requests <= Pod level limits is valid<br>
+          Hence, by transitivity these rules will ensure Container level requests <= Pod-level limits.
+
+      4. [Already exists] Sum of container level requests less than container level limits is invalid.
+
+
+#### Scheduler Changes
+
+The scheduler logic needs to be updated to support pod-level requests and limits.
+
+* **Scenario 1: Only container-level resources are specified**
+
+  No changes, same as existing.
+
+* **Scenario 2: Only pod-level resources are specified**
+
+  Consider pod-level requests to find the suitable node for pod placement.
+
+* **Scenario 3: Both container-level and pod-level resources are specified**
+
+  Scheduler will prioritize pod-level requests as the sole indicator of the pod’s resources needs to find a node that will fulfill the overall 
+  resource requirements of the pod. It will ignore container level requests.
+
+#### QoS Changes
```

Comment:
Similar to above, this whole section feels like it could be reduced to something like "QoS is determined by the derived pod resources, which considers the pod-level resource stanza and all the containers' resources stanzas."

Reply:
+1 agree